### PR TITLE
Bluetooth: ISO: Fix issue with bt_iso_cig_reconfigure

### LINKED ISSUE #61361: Bluetooth: ISO: bt_iso_cig_reconfigure cannot be used when the CIG contains an existing CIS
**Describe the bug**
When a CIG already contains a CIS, `bt_iso_cig_reconfigure()` will return `-EINVAL` or `-EIO` depending on if the existing CIS is supplied in the input parameters.

From the API description it is unclear if it is a requirement if the existing CISes shall always be supplied. The implementation does not add such validation.

**Case 1**: The CIS is part of `param`, another CIS is also added
In this configuration `valid_cig_param()` fails as it has the check:
```
if (cis->iso != NULL) {
	LOG_DBG("cis_channels[%d]: already allocated", i);
	return false;
}
```

**Case 2**: The CIS is part of not `param`, another CIS is added
In this configuration, the CIG state is corrupted with invalid handles, as it does not take existing CISes into account.
```
i = 0;
SYS_SLIST_FOR_EACH_CONTAINER(&cig->cis_channels, cis, node) {
	const uint16_t handle = cig_rsp->handle[i++];

	/* Assign the connection handle */
	cis->iso->handle = sys_le16_to_cpu(handle);
}
```
As a result, the handle of exiting CISes is unintendedly overwritten.

**To Reproduce**
1. Call `bt_iso_cig_create()` with a single CIS
2. Call `bt_iso_cig_reconfigure()` with another CIS with or without the existing CIS
3. Try using both CISes with `bt_iso_chan_connect()`

**Expected behavior**
The API can be used. 
The API describes how it is supposed to be used.
The implementation validates that the API is used correctly.

**Impact**
Annoyance. The workaround is to not use this API and always call `bt_iso_cig_terminate()` and `bt_iso_cig_create()` when a reconfiguration is needed.

**Additional context**
It looks like this API is not used anywhere. Maybe it is better to remove it as it has never worked as intented.

### PR DESCRIPTION
**Pull request by Thalley**
bt_iso_cig_reconfigure would almost always fail due to invalid checks that did not take the functioner properly into account.

The CIS provided to bt_iso_cig_reconfigure can now be allocated beforehand, if the CIG for the CIS is the same as the one being reconfigured.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/61361

### CODE REVIEW

File: tests/bsim/bluetooth/host/iso/cis/src/cis_central.c [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	/* Add CIS to the CIG and restore interval to 10ms */
+	param.num_cis = 2U;
+	param.interval = interval_us; /* us */
+
+	err = bt_iso_cig_reconfigure(cig, &param);
+	if (err != 0) {
+		FAIL("Failed to reconfigure CIG with new CIS and original interval (%d)\n", err);
+
+		return;
+	}
```
Comment (rugeGerritsen) [inline tests/bsim/bluetooth/host/iso/cis/src/cis_central.c:234]: We could also consider testing reconfiguring the cig with num_cis set to 0.

**Review by rugeGerritsen — CHANGES_REQUESTED** [review]
I looked at the API `bt_iso_cig_reconfigure()` only.

File: tests/bsim/bluetooth/host/iso/cis/src/cis_central.c (hunk 1 above)
Comment (Thalley) [reply tests/bsim/bluetooth/host/iso/cis/src/cis_central.c:247]: That was a good idea - It uncovered another issue :)

**Review by rugeGerritsen — APPROVED** [review]
Thanks for also adding a test for it

**Review by cvinayak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**